Controller/OneClick/PlaceOrder.php

```diff
@@ -181,7 +181,12 @@ public function execute()
         } else {
             /** @var QuoteBuilder $quoteBuilder */
             $quoteBuilder = $this->quoteBuilderFactory->create();
-            $quote = $quoteBuilder->createQuote();
+            $buyNowCartPushAction = $this->config->getBuyNowAction();
+            if ($buyNowCartPushAction === '1') {
+                $quote = $quoteBuilder->createOrUpdateQuote();
+            } else {
+                $quote = $quoteBuilder->createQuote();
+            }
```

Comment:
This can be simplified to:
```suggestion
            $quote = $this->config->getBuyNowAction()
                ? $quoteBuilder->createOrUpdateQuote()
                : $quoteBuilder->createQuote();
```